activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -148,6 +148,37 @@ def build_explain_clause(options = [])
           "EXPLAIN (#{options.join(", ").upcase})"
         end
 
+        # Set when constraints will be checked for the current transaction.
+        #
+        # Not passing any specific constraint names will set the value for all deferrable constraints.
+        #
+        # [<tt>deferred</tt>]
+        #   Valid values are +:deferred+ or +:immediate+.
+        # See https://www.postgresql.org/docs/current/sql-set-constraints.html
+        def set_constraints(deferred, *constraints)
+          unless %i[deferred immediate].include?(deferred)
+            raise ArgumentError, "deferred must be :deferred or :immediate"
+          end
+
+          constraints = if constraints.empty?
+            "ALL" if constraints.empty?
```

Comment:
Why `constraints` will change from empty to not empty between 163 and 164?

Why the double `if` here?

Reply:
Heh. Cause I refactored how this was structured. Will fix

Reply:
Fixed